apps/coordinator/ts/deployer/__tests__/deployer.service.test.ts

```diff
@@ -1,148 +1,190 @@
+  } as unknown as FileService;
 
-  const fileService = new FileService();
+  const sessionKeyService = {
+    getCoordinatorSigner: jest.fn().mockResolvedValue(signer),
+  } as unknown as SessionKeysService;
 
-  const storageInstance = ContractStorage.getInstance(path.join(process.cwd(), "deployed-contracts.json"));
-  const sessionKeyService = new SessionKeysService(fileService);
-  const deployerService = new DeployerService(sessionKeyService, fileService);
+  beforeEach(() => {
+    mockStorage.getAddress = jest.fn().mockReturnValue(zeroAddress);
 
-  let approval: string;
-  let sessionKeyAddress: Hex;
-  let signer: Signer;
+    sessionKeyService.getCoordinatorSigner = jest.fn().mockResolvedValue(signer);
 
-  beforeAll(async () => {
-    sessionKeyAddress = (await sessionKeyService.generateSessionKey()).sessionKeyAddress;
-    approval = await generateApproval(sessionKeyAddress);
+    fileService.getZkeyFilePaths = jest.fn().mockReturnValue({ zkey: "" });
+
+    (ContractStorage.getInstance as jest.Mock).mockReturnValue(mockStorage);
+
+    (deployFreeForAllSignUpPolicy as jest.Mock).mockResolvedValue([
+      mockContract,
+      mockContract,
+      mockContract,
+      mockContract,
+    ]);
+
+    (deployVerifier as jest.Mock).mockResolvedValue(mockContract);
+
+    (deployVerifyingKeysRegistryContract as jest.Mock).mockResolvedValue(mockContract);
+
+    (deployConstantInitialVoiceCreditProxyFactory as jest.Mock).mockResolvedValue(mockContract);
+
+    (deployConstantInitialVoiceCreditProxy as jest.Mock).mockResolvedValue(mockContract);
+
+    (deployPoll as jest.Mock).mockResolvedValue({
+      pollContractAddress: zeroAddress,
+      messageProcessorContractAddress: zeroAddress,
+      tallyContractAddress: zeroAddress,
+      pollId: 0n,
+    });
+
+    (deployMaci as jest.Mock).mockResolvedValue({
+      maciContractAddress: zeroAddress.replace("0x0", "0x1"),
+      pollFactoryContractAddress: zeroAddress,
+      messageProcessorFactoryContractAddress: zeroAddress,
+      tallyFactoryContractAddress: zeroAddress,
+      poseidonAddresses: {
+        poseidonT3: zeroAddress,
+        poseidonT4: zeroAddress,
+        poseidonT5: zeroAddress,
+        poseidonT6: zeroAddress,
+      },
+    });
 
-    signer = await sessionKeyService.getCoordinatorSigner(chain);
+    (getDeployedPolicyProxyFactories as jest.Mock).mockResolvedValue({ checker: undefined, policy: undefined });
   });
 
-  afterAll(() => {
-    storageInstance.cleanup(ESupportedNetworks.OPTIMISM_SEPOLIA);
-    storageInstance.cleanup(ESupportedNetworks.ARBITRUM_ONE);
+  afterEach(() => {
+    jest.clearAllMocks();
   });
 
   describe("deployAndSavePolicy", () => {
-    // we cleanup after each test so we don't have leftover saved contracts
-    afterEach(() => {
-      storageInstance.cleanup(chain);
-    });
-
     test("should throw when the policy is not existent", async () => {
+      const deployerService = new DeployerService(sessionKeyService, fileService);
```

Comment:
why are we declaring the `deployerService` in the tests and not as a variable on top so all tests use it?

Reply:
Because mocks should be added before initialization of deployer service